programs/whirlpool/src/state/position.rs

```diff
@@ -69,6 +69,16 @@ impl Position {
             return Err(ErrorCode::InvalidTickIndex.into());
         }
 
+        // On tick spacing >= 2^15, should only be able to open full range positions (infinity pool)
+        if whirlpool.tick_spacing >= 2u16.pow(15) {
+            // Use gt or lt here instead of neq since indexes still need to be on tick spacing
+            if tick_lower_index < MIN_TICK_INDEX
+                || tick_upper_index > MAX_TICK_INDEX
+            {
+                return Err(ErrorCode::InvalidTickIndex.into());
+            }
+        }
```

Comment:
How about changing `2u16.pow(15)`  to constant (pre calculated).

I guess this condition will not reject `[-32768, +32768]` and `[-32768, 0]`, etc.
I think we need to calculate full range lower index & upper index based on tick spacing.

Adding an error code indicating "FullRange only" would be better for the developer to know the cause.

Reply:
Ahh yes. Changed this because I didn't want to have to calculate the exact full range indexes to do an eq check (initially thought this would get an identical result but is not the case)